examples/widget_list.rs

```diff
@@ -0,0 +1,710 @@
+use crossterm::{
+    event::{self, DisableMouseCapture, EnableMouseCapture, Event, KeyCode, KeyEventKind},
+    execute,
+    terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
+};
+use ratatui::{
+    backend::{Backend, CrosstermBackend},
+    layout::{Constraint, Direction, Layout, Rect},
+    style::{Color, Modifier, Style},
+    text::{Span, Spans},
+    widgets::{
+        BarChart, Block, Borders, Gauge, ListItem, Paragraph, WidgetList, WidgetListItem3,
+        WidgetListState, Wrap,
+    },
+    Frame, Terminal,
+};
+use std::{
+    error::Error,
+    io,
+    time::{Duration, Instant},
+};
+
+struct StatefulList<T> {
+    state: WidgetListState,
+    items: Vec<T>,
+}
+
+impl<T> StatefulList<T> {
+    fn with_items(items: Vec<T>) -> StatefulList<T> {
+        StatefulList {
+            state: WidgetListState::default(),
+            items,
+        }
+    }
+
+    fn next(&mut self) {
```

Comment:
I think we should probably fold `next()` and `previous()` directly into the ListState as `select_next()/select_prev()`, and that that approach could be done outside this change. Does that sound like a good idea to you?

Reply:
yes I think, this is an improvement